src/multimap.js

```diff
@@ -0,0 +1,21 @@
+// @flow
+import { List } from 'immutable';
+
+export default class Multimap<K, V> extends Map<K, List<V>> {
+  add(key: K, value: V) {
+    let bucket = this.get(key);
+    if (bucket != null) {
+      this.set(key, bucket.concat(value));
```

Comment:
Should this be `bucket.push(value)`? `value` is `phase` (a number) in this instance. Correct?